tools/build/tests/devfile-yaml/index-writer.spec.ts

```diff
@@ -125,5 +115,80 @@ describe('Test IndexWriter', () => {
     expect(jsonOutput[2].publisher).toBe('my-publisher');
     expect(jsonOutput[2].type).toBe('Che Editor');
     expect(jsonOutput[2].version).toBe('latest');
+    expect(jsonOutput[2].icon).toBe('/v3/images/vscode.svg');
+  });
+
+  test('throw an error if metadata is empty', async () => {
+    editors = [
+      {
+        schemaVersion: '2.2.2',
+      },
+    ];
+
+    const fsEnsureDirSpy = jest.spyOn(fs, 'ensureDir');
+    const fsWriteFileSpy = jest.spyOn(fs, 'writeFile');
+
+    fsEnsureDirSpy.mockReturnValue();
+    fsWriteFileSpy.mockReturnValue();
+
+    let error: any;
+    try {
+      await indexWriter.write(editors);
+    } catch (e: any) {
+      error = e;
+    }
+    expect(error.message).toBe('The metadata of {"schemaVersion":"2.2.2"} is not defined');
```

Comment:
Although this test will work correctly (as long as the `error` variable is defined), I would like to suggest two alternatives for testing this asynchronous code.

The first one uses the `rejects` modifier (see [Modifiers](https://jestjs.io/docs/expect#modifiers)):
```ts
await expect(indexWriter.write(editors)).rejects.toThrow('The metadata of {"schemaVersion":"2.2.2"} is not defined');
```

And the second one uses the `expect.assertions` (see [Assertion Count](https://jestjs.io/docs/expect#assertion-count)):
```ts
expect.assertions(1);
try {
  await indexWriter.write(editors);
} catch (e) {
  expect(e.message).toBe('The metadata of {"schemaVersion":"2.2.2"} is not defined');
}
```